src/utils/cardUtilityFunctions.js

```diff
@@ -122,3 +124,403 @@ export const getUpdatedCardSize = oldSize => {
   }
   return newSize;
 };
+
+// re-usable variable fetcher that returns an array of variables
+// variables are identified by surrounding curly braces i.e. {deviceid}
+export const getVariables = value => {
```

Comment:
Could you provide comments in jsdoc style for this and the other functions? The logic is quite dense and would be beneficial to have some specifics on return values, param types, etc.

```suggestion
/**
 * Find variables that match a given value, identified by surrounding curly braces, i.e. `{deviceid}`
 * @param {string} value - The description of the value
 * @return {array} variables - The description of the variables
 */
export const getVariables = value => {
```